### PR TITLE
Completed html-forms exercise

### PR DESCRIPTION
**Pull request by DavidKyong**
Screenshot 2023-05-15 at 7 57 27 PM

### CODE REVIEW

File: html-forms/index.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      HTML Forms
+    </title>
+  </head>
+  <body>
+    <h1>Coding Form!</h1>
+    <form>
+<div>
+          <div>Email</div>
+          <div>
+            <label for="email"></label>
```
Comment (shawnkost) [inline html-forms/index.html]: Your "Email" text on line 12 should be inside this `label`

File: html-forms/index.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            <input type="email" id="email" name="usermail" />
+          </div>
+
+        <div>Why is coding cool?</div>
+        <div>
+      <select id="card" name="dropdown">
```
Comment (shawnkost) [inline html-forms/index.html]: This should be a `label` & it should be attached to your `select` on line 20

File: html-forms/index.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        <option value="option"><selected>Select an option</selected></option>
+        <option value="interesting">Because it's interesting</option>
+        <option value="fun">Because it is fun</option>
+        <option value="powerful">Because it is powerful</option>
+      </select>
+      </div>
+    <p>Favorite Language (choose one)</p>
+    <div>
+          <label for="firstcheck"></label>
+       <input type="radio" id="firstcheck" name="check" value="html" /> HTML
```
Comment (shawnkost) [inline html-forms/index.html]: Your HTML text should be inside the `label`. You can just put the `input` inside the `label`

File: html-forms/index.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    <label for="secondcheck"></label>
+        <input type="radio" id="secondcheck" name="check" value="css" /> CSS
```
Comment (shawnkost) [inline html-forms/index.html]: Same note as above ^

File: html-forms/index.html [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+     <label for="thirdcheck"></label>
+    <input type="radio" id="thirdcheck" name="check" value="javascript" /> JavaScript
```
Comment (shawnkost) [inline html-forms/index.html]: Same note as above ^

File: html-forms/index.html [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+</div>
+    <p>Would you like to sign up for more information?</p>
+
+<div>
+ <input type="checkbox" id="sign" name="sign" value="signup"/> Sign me up!
```
Comment (shawnkost) [inline html-forms/index.html]: The "Sign me up" text should be inside a `label` and it should be linked to this `input`

File: html-forms/index.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+</div>
+ <div>Comments</div>
```
Comment (shawnkost) [inline html-forms/index.html]: This should be a `label`, and it should be linked to your `textarea`

**Review by shawnkost — CHANGES_REQUESTED** [review]
Some changes requested here David!

File: html-forms/index.html [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+          <div>
+            <label for="email">
+             <div>Email</div>
+            <input type="email" id="email" name="usermail" />
+            </label>
+          </div>
+
+        <div>Why is coding cool?</div>
+        <div>
+      <label for="card"></label>
```
Comment (shawnkost) [inline html-forms/index.html]: The text inside your `div` on line 19 should be inside your `label` on line 21

File: html-forms/index.html [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+<div>
+  <label for="sign">
+ <input type="checkbox" id="sign" name="sign" value="signup"/> Sign me up!
+ </label>
+</div>
+ <div>Comments</div>
+ <div>
+  <label for="box">
+  <textarea cols="20" rows="10" id="box"></textarea>
+  </label>
```
Comment (shawnkost) [inline html-forms/index.html]: The text inside your `div` on line 48 should be inside your `label` on line 50

**Review by shawnkost — CHANGES_REQUESTED** [review]
A couple changes here David!

File: html-forms/index.html [hunk 10]
```diff
@@ -1,60 +1,63 @@
...
-             <div>Email</div>
-            <input type="email" id="email" name="usermail" />
-            </label>
-          </div>
+      <div>
+        <div>
+          <label for="email"><div>Email</div>
+          <input type="email" id="email" name="usermail" />
+          </label>
+        </div>
```
Comment (shawnkost) [inline html-forms/index.html]: You can't put a `div` inside a `label`. The only thing inside a `label` should be the text itself and an `input` or some sort of form control element.

File: html-forms/index.html [hunk 11]
```diff
@@ -1,60 +1,63 @@
...
-      <select id="card" name="dropdown">
-        <option value="option"><selected>Select an option</selected></option>
-        <option value="interesting">Because it's interesting</option>
-        <option value="fun">Because it is fun</option>
-        <option value="powerful">Because it is powerful</option>
-      </select>
-      </div>
-    <p>Favorite Language (choose one)</p>
-    <div>
+          <label for="card"><div>Why is coding cool?</div>
```
Comment (shawnkost) [inline html-forms/index.html]: You can't put a `div` inside a `label`. The only thing inside a `label` should be the text itself and an `input` or some sort of form control element.

File: html-forms/index.html [hunk 12]
```diff
@@ -1,60 +1,63 @@
...
+          <select id="card" name="dropdown">
+            <option value="option" selected>Select an option</option>
+            <option value="interesting">Because it's interesting</option>
+            <option value="fun">Because it is fun</option>
+            <option value="powerful">Because it is powerful</option>
+          </select>
+          </label>
```
Comment (shawnkost) [inline html-forms/index.html]: It's either you use the `for` attribute, OR you nest the form control element inside the `label`. Not both. It's one or the other

File: html-forms/index.html [hunk 13]
```diff
@@ -1,60 +1,63 @@
...
-            <label for="email">
-             <div>Email</div>
-            <input type="email" id="email" name="usermail" />
-            </label>
-          </div>
+      <div>
+        <div>
+          <label for="email"><div>Email</div>
+          <input type="email" id="email" name="usermail" />
+          </label>
```
Comment (shawnkost) [inline html-forms/index.html]: It's either you use the `for` attribute, OR you nest the form control element inside the `label`. Not both. It's one or the other

File: html-forms/index.html [hunk 14]
```diff
@@ -1,60 +1,63 @@
...
-    <label for="secondcheck">
-        <input type="radio" id="secondcheck" name="check" value="css" /> CSS
-      </label>
-     <label for="thirdcheck">
-    <input type="radio" id="thirdcheck" name="check" value="javascript" /> JavaScript
-    </label>
-</div>
-    <p>Would you like to sign up for more information?</p>
+            <input type="radio" id="firstcheck" name="check" value="html" />
+            <span>HTML</span>
```
Comment (shawnkost) [inline html-forms/index.html]: No need for the `span`, just have the text in here directly

File: html-forms/index.html [hunk 15]
```diff
@@ -1,60 +1,63 @@
...
+          </label>
+          <label for="secondcheck">
+            <input type="radio" id="secondcheck" name="check" value="css" />
+            <span>CSS</span>
```
Comment (shawnkost) [inline html-forms/index.html]: No need for the `span`, just have the text in here directly

File: html-forms/index.html [hunk 16]
```diff
@@ -1,60 +1,63 @@
...
+          </label>
+          <label for="thirdcheck">
+            <input type="radio" id="thirdcheck" name="check" value="javascript" />
+            <span>JavaScript</span>
```
Comment (shawnkost) [inline html-forms/index.html]: No need for the `span`, just have the text in here directly

File: html-forms/index.html [hunk 17]
```diff
@@ -1,60 +1,63 @@
...
-    <div>
-    <button type="submit">Submit</button>
-    </div>
-</div>
-  </form>
+        <p>Would you like to sign up for more information?</p>
+        <div>
+          <label for="sign">
+            <input type="checkbox" id="sign" name="sign" value="signup" />
+            <span>Sign me up!</span>
```
Comment (shawnkost) [inline html-forms/index.html]: No need for the `span`, just have the text in here directly

File: html-forms/index.html [hunk 18]
```diff
@@ -1,60 +1,63 @@
...
+          </label>
+        </div>
+        <div>
+          <label for="box"> <div>Comments</div>
```
Comment (shawnkost) [inline html-forms/index.html]: You can't put a `div` inside a `label`. The only thing inside a `label` should be the text itself and an `input` or some sort of form control element.

**Review by shawnkost — CHANGES_REQUESTED** [review]
Some more changes requested here David. A few things

1. The only content that should ever be inside a `label`, is text, or an `input`, `select`, etc. You can't put `div` elements or `span` elements
2. You either use the `for` attribute to connect `label`s to form control elements, OR you nest them inside the `label`, but you don't do both. It's one or the other.

Make sure you update my comments for this whole file.

**Comment by DavidKyong** [comment]
Sorry, I accidentally commit it when I was not completed

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**